### ISSUE TITLE
Non-zero status code returned while running Conv node

### ISSUE THREAD

**Issue by hoaquocphan** [onnx/tensorflow-onnx#1062]
I converted the saved model **saved_model.pb** created from https://github.com/atomicbits/posenet-python to onnx model with below command:
`python3 -m tf2onnx.convert --saved-model /mobilenet_float_100/stride16 --output mobilenet_stride16.onnx`

when load onnx model, it can print out input,output size,dim correctly,
but when run it, it have the error as below:

> (base) hp@UBT16:/media/sf_Ubuntu/posenet/posenet_demo$ ./PoseNetDemo -model mobilenet -stride 16 -ofile "./output/out.png" -quant_bytes 4 -ifile "./images/tie_with_beer.jpg"
> Start Loading Model mobilenet
> tget_wid: 513 
> stride: 16 
> arr_size: 33 
> 
> Current Model is mobilenet
> Number of inputs = 1
> Number of outputs = 4
> Input 0 : name=sub_2:0
> Input 0 : type=1
> Input 0 : num_dims=4
> Input 0 : dim 0=1
> Input 0 : dim 1=513
> Input 0 : dim 2=513
> Input 0 : dim 3=3
> output 0 : name=MobilenetV1/displacement_bwd_2/BiasAdd:0
> Output : output_name=MobilenetV1/displacement_bwd_2/BiasAdd:0
> Output 0 : type=1
> Output 0 : num_dims=4
> output 0 : dim 0=1
> output 0 : dim 1=33
> output 0 : dim 2=33
> output 0 : dim 3=32
> output 1 : name=MobilenetV1/displacement_fwd_2/BiasAdd:0
> Output : output_name=MobilenetV1/displacement_fwd_2/BiasAdd:0
> Output 1 : type=1
> Output 1 : num_dims=4
> output 1 : dim 0=1
> output 1 : dim 1=33
> output 1 : dim 2=33
> output 1 : dim 3=32
> output 2 : name=MobilenetV1/heatmap_2/BiasAdd:0
> Output : output_name=MobilenetV1/heatmap_2/BiasAdd:0
> Output 2 : type=1
> Output 2 : num_dims=4
> output 2 : dim 0=1
> output 2 : dim 1=33
> output 2 : dim 2=33
> output 2 : dim 3=17
> output 3 : name=MobilenetV1/offset_2/BiasAdd:0
> Output : output_name=MobilenetV1/offset_2/BiasAdd:0
> Output 3 : type=1
> Output 3 : num_dims=4
> output 3 : dim 0=1
> output 3 : dim 1=33
> output 3 : dim 2=33
> output 3 : dim 3=34
> 2020-08-13 08:09:45.107916526 [E:onnxruntime:, sequential_executor.cc:165 Execute] Non-zero status code returned while running Conv node. Name:'Conv__280' Status Message: /data/onnxruntime/onnxruntime/core/providers/cpu/nn/conv_attributes.h:38 onnxruntime::common::Status onnxruntime::ComputePadAndOutputShape(int64_t, int64_t, int64_t, int64_t, onnxruntime::AutoPadType, int64_t*, int64_t*, int64_t*) [with bool ForceSymmetricAutoPadding = false; int64_t = long int] dilation == 1 was false. Dilation not supported for AutoPadType::SAME_UPPER or AutoPadType::SAME_LOWER.
> Stacktrace:
> 
> Non-zero status code returned while running Conv node. Name:'Conv__280' Status Message: /data/onnxruntime/onnxruntime/core/providers/cpu/nn/conv_attributes.h:38 onnxruntime::common::Status onnxruntime::ComputePadAndOutputShape(int64_t, int64_t, int64_t, int64_t, onnxruntime::AutoPadType, int64_t*, int64_t*, int64_t*) [with bool ForceSymmetricAutoPadding = false; int64_t = long int] dilation == 1 was false. Dilation not supported for AutoPadType::SAME_UPPER or AutoPadType::SAME_LOWER.
> Stacktrace:

So could anyone help me to fix this issue?
Thansk

**Comment by guschmue** [comment]
what version of tf2onnx and onnxruntime was this?

**Comment by hoaquocphan** [comment]
Hi @guschmue 
Thanks for your reply.

the version of onnxruntime is 1.0.0
about  tf2onnx:
I clone the master branch with commit below:

```
commit 0110037896b2a95654f9e79fecf7a8092f8cc20b
Merge: c867e52 3ead413
Author: TomWildenhain-Microsoft <67606533+TomWildenhain-Microsoft@users.noreply.github.com>
Date:   Mon Aug 10 13:28:59 2020 -0400

    Merge pull request #1050 from onnx/tom/FixQueueDequeueUpToV2Error
    
    Added brackets to fix error in QueueDequeueUpToV2
```
I guess it is tf2onnx-v1.6.3

**Comment by guschmue** [comment]
Hm, onnxruntime-1.0 is really old. Maybe try onnxruntime-1.4 ... lots of fixes when into there. If that still fails we'll look into tf2onnx.

**Comment by hoaquocphan** [comment]
Hi @guschmue 

I just follow your guide:
1/ build onnxruntime 1.4.0
```
6/6 Test #6: onnxruntime_api_tests_without_env ......   Passed    0.01 sec

100% tests passed, 0 tests failed out of 6

Total Test time (real) =  48.94 sec
2020-08-13 11:07:30,767 Build [DEBUG] - Subprocess completed. Return code=0
2020-08-13 11:07:30,768 Build [INFO] - Build complete
```

2/ re-run my app to run onnx model after converting, the error still occur same and it also have more error as below:

> 2020-08-13 11:20:47.539823737 [W:onnxruntime:, graph.cc:863 Graph] Initializer Const_1:0 appears in graph inputs and will not be treated as constant value/weight. This may prevent some of the graph optimizations, like const folding. Move it out of graph inputs if there is no need to override it, by either re-generating the model with latest exporter/converter or with the tool onnxruntime/tools/python/remove_initializer_from_input.py.
> 2020-08-13 11:20:47.539844702 [W:onnxruntime:, graph.cc:863 Graph] Initializer Const:0 appears in graph inputs and will not be treated as constant value/weight. This may prevent some of the graph optimizations, like const folding. Move it out of graph inputs if there is no need to override it, by either re-generating the model with latest exporter/converter or with the tool onnxruntime/tools/python/remove_initializer_from_input.py.
> Start Loading Model mobilenet
> tget_wid: 513 
> stride: 16 
> arr_size: 33 
> 
> Current Model is mobilenet
> Number of inputs = 1
> Number of outputs = 4
> Input 0 : name=sub_2:0
> Input 0 : type=1
> Input 0 : num_dims=4
> Input 0 : dim 0=1
> Input 0 : dim 1=513
> Input 0 : dim 2=513
> Input 0 : dim 3=3
> output 0 : name=MobilenetV1/displacement_bwd_2/BiasAdd:0
> Output 0 : type=1
> Output 0 : num_dims=4
> output 0 : dim 0=1
> output 0 : dim 1=33
> output 0 : dim 2=33
> output 0 : dim 3=32
> output 1 : name=MobilenetV1/displacement_fwd_2/BiasAdd:0
> Output 1 : type=1
> Output 1 : num_dims=4
> output 1 : dim 0=1
> output 1 : dim 1=33
> output 1 : dim 2=33
> output 1 : dim 3=32
> output 2 : name=MobilenetV1/heatmap_2/BiasAdd:0
> Output 2 : type=1
> Output 2 : num_dims=4
> output 2 : dim 0=1
> output 2 : dim 1=33
> output 2 : dim 2=33
> output 2 : dim 3=17
> output 3 : name=MobilenetV1/offset_2/BiasAdd:0
> Output 3 : type=1
> Output 3 : num_dims=4
> output 3 : dim 0=1
> output 3 : dim 1=33
> output 3 : dim 2=33
> output 3 : dim 3=34
> 2020-08-13 11:20:48.337098243 [E:onnxruntime:, sequential_executor.cc:309 Execute] Non-zero status code returned while running FusedConv node. Name:'fused Conv__280' Status Message: /data/onnxruntime/onnxruntime/core/providers/common.h:95 onnxruntime::common::Status onnxruntime::ComputePadAndOutputShape(int64_t, int64_t, int64_t, int64_t, onnxruntime::AutoPadType, int64_t*, int64_t*, int64_t*) [with bool ForceSymmetricAutoPadding = false; int64_t = long int] dilation == 1 was false. Dilation not supported for AutoPadType::SAME_UPPER or AutoPadType::SAME_LOWER.
> 
> Non-zero status code returned while running FusedConv node. Name:'fused Conv__280' Status Message: /data/onnxruntime/onnxruntime/core/providers/common.h:95 onnxruntime::common::Status onnxruntime::ComputePadAndOutputShape(int64_t, int64_t, int64_t, int64_t, onnxruntime::AutoPadType, int64_t*, int64_t*, int64_t*) [with bool ForceSymmetricAutoPadding = false; int64_t = long int] dilation == 1 was false. Dilation not supported for AutoPadType::SAME_UPPER or AutoPadType::SAME_LOWER.
> 


So could you check and give me your support?
Thanks

**Comment by TomWildenhain-Microsoft** [comment]
It looks like there's a conv node with dilation > 1.  During conversion, we don't use autopadding unless the input dimensions to the model are unknown.  If the model really does have a conv node with unknown input dimensions and dilation > 1, it looks like ONNX runtime can't run it yet.  Otherwise, there's a bug in the converter.

**Comment by TomWildenhain-Microsoft** [comment]
Let me know if you still need help with this model.

**Comment by hoaquocphan** [comment]
Hi @TomWildenhain-Microsoft 

Thanks for your reply
this issue still exist. but currently, I used other model to run the Posenet app.

**Comment by TomWildenhain-Microsoft** [comment]
@hoaquocphan Cool.  Would you be able to upload both the trained TF saved model and the converted onnx file?  I'll take a look and see if there's any way we can change the conversion to get ORT to accept it.  Also are you using the latest version of ORT?  If there is nothing we can do, we can open an issue with ORT.

**Comment by tsedlmeier** [comment]
Hey Guys,
i have a similar Problem with my SSD-Mobilenet v2 on Jetson Nano.
The ONNXRuntime Inference performs at ~2FPS throwing some warnings like "Unsupported Layer Fallback to CPU"
However the TensorRT Inference fails bc of Unsupported Plugin NonZero

[inference.zip](https://github.com/onnx/tensorflow-onnx/files/5943999/inference.zip)
Heres my Project maybe someone can reproduce the Issues 
Thanks a lot

**Comment by tsedlmeier** [comment]
[convert_console_log.txt](https://github.com/onnx/tensorflow-onnx/files/5944039/convert_console_log.txt)
I used tf2onnx script, which logged me this messages.

**Comment by TomWildenhain-Microsoft** [comment]
Hi @tsedlmeier, your issue is caused by Jetson not implementing the NonZero op.  Your model contains this op in two places (within an loop and an if node).  They come from a TF Where op that is somehow related to "boolean masks".  If you can remove that op from your model or find a different way to implement it in TF, the model should convert.

![image](https://user-images.githubusercontent.com/67606533/107280220-07608680-6a26-11eb-99ca-4dd49757cd27.png)

**Comment by tsedlmeier** [comment]
Thank you for your hint. You are absolutly right... these "Where" OPs occure in my original TF model.
However i dont know if they are necessary  and how to specifically remove them.
Maybe you can have a look at my TF model.
[frozen_graph_infer.zip](https://github.com/onnx/tensorflow-onnx/files/5949623/frozen_graph_infer.zip)

**Comment by TomWildenhain-Microsoft** [comment]
To most easily remove the Where ops you will probably need to modify the original TF python code.  Alternatively, you could try to implement the behavior of the onnx NonZero op using other ops that Jetson does support. It's a bit of a Tensor logic puzzle.  Once you have an alternative implementation, it can be added to the converter and used when targeting Jetson.  You could also manually edit the model to replace the op with your implementation.

Do you have to use Jetson Nano (not ONNX runtime)?  onnx runtime calls Jetson as an execution provider, so it should be just as fast.  If ORT is slower than TF, you could open an issue with ORT: https://github.com/microsoft/onnxruntime/issues

**Comment by TomWildenhain-Microsoft** [comment]
Correction: should have said "TensorRT" not "Jetson"

**Comment by tsedlmeier** [comment]
Thank you. I assume that editing the model is the easyest way of getting rid of the problem.
The reason for using TensorRT is because of ORT beeing slow (~2FPS) in my case and Nvidia Developer advides me to better use TensorRT.
But i parallely will open an Issue on ORTs Github page.
One question left: How can i add the model after training? A collegue trained the model with his own ecosystem where i dont have any access.
Is there a way or tool for this case?

**Comment by TomWildenhain-Microsoft** [comment]
"How can i add the model after training?" - Don't think I understand.  Do you mean edit the model after training?  To edit the model on the TF side, you will need the TF source used to generate the model, edit the python, and make a new model.  You can copy the weights from the trained model to the new saved_model.

**Comment by tsedlmeier** [comment]
Thanks for your reply, 
i got the Inferece with TensorRT working by editing the model ("cutting" unsupported layer ) and depicting the functionality in software.

**Comment by hoaquocphan** [comment]
Hi @TomWildenhain-Microsoft 

I convert model and have the error with Squeeze node, but the error log are same:
`onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running Squeeze node. Name:'StatefulPartitionedCall/StatefulPartitionedCall/predict/MobilenetV2/Logits/Squeeze' Status Message: /onnxruntime_src/onnxruntime/core/providers/common.h:18 int64_t onnxruntime::HandleNegativeAxis(int64_t, int64_t) axis >= -tensor_rank && axis <= tensor_rank - 1 was false. axis 2 is not in valid range [-2,1]`

could you have a check and give me your support?

**Comment by TomWildenhain-Microsoft** [comment]
Hi @hoaquocphan, that looks like a very different error (all onnx runtime errors start with "non-zero status code").  Can you please open a new issue and upload the model you are converting?

**Comment by hoaquocphan** [comment]
Hi @TomWildenhain-Microsoft 

I create the new ticket here: https://github.com/onnx/tensorflow-onnx/issues/1418 
could you have a check and give me the support

Thank you very much!

**Comment by snnn** [comment]
> It looks like there's a conv node with dilation > 1. During conversion, we don't use autopadding unless the input dimensions to the model are unknown. If the model really does have a conv node with unknown input dimensions and dilation > 1, it looks like ONNX runtime can't run it yet. Otherwise, there's a bug in the converter.

I have the same question: does it need be supported in ONNX Runtime or this converter? Have you ever found such a model in TF?  I couldn't. ref:  https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/framework/common_shape_fns.cc#L35
